biz.aQute.bndlib/src/aQute/bnd/osgi/Macro.java

```diff
@@ -356,6 +354,24 @@ private String doCommand(Object target, String method, String[] args) {
 		return null;
 	}
 
+	private Method lookupMacroMethod(String method, Object target) {
+		// Assume macro names do not start with '-' and does not contain '.'
+		if (method.startsWith("-") || method.indexOf('.') > -1) {
+			return null;
+		}
+		String cname = '_' + method.replace('-', '_');
+		for (Method m : target.getClass().getMethods()) {
```

Comment:
I just realized that this has a problem with inheritance. getMethod(String,Class[]) returns the overridden method. When you use getMethods, all the overrloads are in the result in no specific order. And you loop just uses the first method found which can be from a superclass. This wrong.

If you want to use getMethods, you will need to find all the matching method and the pick the method from the type closest to the target's class.